### ISSUE TITLE
How to deal with unmatched sample size?

### ISSUE THREAD

**Issue by LiJiaqi96** [txWang/BERMUDA#4]
Hi,

Thanks for your idea and work. I would like to figure out how did you deal with unmatched sample size, in other words, two batch sizes are not the same. 
I notice that in the MMD loss you used "assert" to ignore this situation. But is there any solution to this problem as the number of samples in two clusters may not be the same.

Thanks!

**Comment by txWang** [comment]
Hi,
Thank you for your question. We apply MMD loss on pairs of similar clusters from different batches. During the construction of a mini-batch for training the network, we sample the same amount of cells from each cell cluster, so the batch size for calculating MMD is always matched. If the size of some cell cluster is insufficient, we simply random over-sample the cells. You could also try to apply MMD for different batch sizes. Here, we just did sample the same amount of cells in mini-batch gradient descent for simplicity.

**Comment by LiJiaqi96** [comment]
Thanks for your in-time reply. It really helps.
